src/main/kotlin/logic/usecase/auth/AuthenticateUserLoginUseCase.kt

```diff
@@ -7,6 +7,15 @@ class AuthenticateUserLoginUseCase(
     private val authRepo: AuthenticationRepository
 ) {
     operator fun invoke(username: String, password: String): Result<User> {
+        if (username.isBlank()) {
+            return Result.failure(IllegalArgumentException("Username cannot be empty"))
+        }
+        if (password.isBlank()) {
+            return Result.failure(IllegalArgumentException("Password cannot be empty"))
+        }
```

Comment:
This should be in the UI instead, as it's input validation, not username and password validation.

Kindly refactor `LoginView` to include this instead.